src/Modules/LearningTasks/Tutor.LearningTasks.Core/Domain/Step.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Tutor.BuildingBlocks.Core.Domain;
+using Tutor.LearningTasks.Core.Domain.Activites;
+
+namespace Tutor.LearningTasks.Core.Domain;
+
+public class Step : Entity
```

Comment:
We need to see where the consistency boundary lies for our aggregates. A few heuristics can help, summarized [here](https://github.com/psw-ftn/supportive-information/tree/master/s2/w1-w2#1-kreiranje-modela-takti%C4%8Dkih-ddd-%C5%A1ablona-za-kontekst-kartice).

I would imagine `Step` is a value object with the `ActivityId` and possibly `ActivityName` if we want to display it when viewing learning tasks (e.g., in the overview card and at the side bar that has the step numbers (the name can be showed on hover). The reason is that an activity has a lifecycle outside a particular learning task aggregate and we only want to load the activity when the learner is on its step (especially given that each activity can take minutes or hours to complete).

In essence, what we wanted to do with the colors on draw.io was to show the different aggregates. Not sure if we are 100% correct, but this should be your starting point.

Reply:
Ok, `Step` will be value object with `ActivityId` and `ActivityName`.